File: 진우/8weeks/5073. 삼각형과 세 변/삼각형과 세 변.py
while True:
    a, b, c = map(int, input().split())

    if a == 0 and b == 0 and c == 0:
        break

    mx = max(a, b, c)
    total = a + b + c - mx

    if mx >= total:
        print("Invalid")  # 삼각형이 될 수 없음
    elif a == b == c:
        print("Equilateral")
    elif a == b or b == c or a == c:
        print("Isosceles")
    else:
        print("Scalene")

File: 진우/8weeks/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    while (true) {
        int a, b, c;
        cin >> a >> b >> c;

        if (a == 0 && b == 0 && c == 0)
            break;

        int mx = max({ a, b, c });
        int sum = a + b + c - mx;

        if (mx >= sum) {
            cout << "Invalid" << "\n";  // 삼각형 불가능
        }
        else if (a == b && b == c) {
            cout << "Equilateral" << "\n";
        }
        else if (a == b || b == c || a == c) {
            cout << "Isosceles" << "\n";
        }
        else {
            cout << "Scalene" << "\n";
        }
    }

    return 0;
}
